EpsilonCCS/Src/CanParsers/MotorCanParser.c

```diff
@@ -0,0 +1,252 @@
+#include "cmsis_os.h"
+
+#include "MotorCanParser.h"
+
+#include "KeyMotorData.h"
+#include "MotorDetailsData.h"
+#include "MotorFaultsData.h"
+
+extern struct KeyMotorData keyMotorData;
+extern struct MotorDetailsData motor0DetailsData;
+extern struct MotorDetailsData motor1DetailsData;
+extern struct MotorFaultsData motorFaultsData;
+
+#define M0_ID (0)
+#define M1_ID (1)
+
+#define M0_BASE_CAN_ID (0x400)
+#define M0_STATUS_CAN_ID (0x401)
+#define M0_BUS_CAN_ID (0x402)
+#define M0_VELOCITY_CAN_ID (0x403)
+#define M0_PHASE_CURRENT_CAN_ID (0x404)
+#define M0_VOLTAGE_VECTOR_CAN_ID (0x405)
+#define M0_CURRENT_VECTOR_CAN_ID (0x406)
+#define M0_BACKEMF_CAN_ID (0x407)
+#define M0_15V_VOLTAGE_RAIL_CAN_ID (0x408)
+#define M0_3_3V_1_9V_VOLTAGE_RAIL_CAN_ID (0x409)
+#define M0_HEATSINK_TEMPERATURE_CAN_ID (0x40B)
+#define M0_DSP_BOARD_TEMPERATURE_CAN_ID (0x40C)
+#define M0_ODOMETER_BUS_AMPHOURS_CAN_ID (0x40E)
+#define M0_SLIP_SPEED_CAN_ID (0x417)
+#define M1_BASE_CAN_ID (0x420)
+#define M1_STATUS_CAN_ID (0x421)
+#define M1_BUS_CAN_ID (0x422)
+#define M1_VELOCITY_CAN_ID (0x423)
+#define M1_PHASE_CURRENT_CAN_ID (0x424)
+#define M1_VOLTAGE_VECTOR_CAN_ID (0x425)
+#define M1_CURRENT_VECTOR_CAN_ID (0x426)
+#define M1_BACKEMF_CAN_ID (0x427)
+#define M1_15V_VOLTAGE_RAIL_CAN_ID (0x428)
+#define M1_3_3V_1_9V_VOLTAGE_RAIL_CAN_ID (0x429)
+#define M1_HEATSINK_TEMPERATURE_CAN_ID (0x42B)
+#define M1_DSP_BOARD_TEMPERATURE_CAN_ID (0x42C)
+#define M1_ODOMETER_BUS_AMPHOURS_CAN_ID (0x42E)
+#define M1_SLIP_SPEED_CAN_ID (0x437)
+
+void parseMotorCanMessage(uint32_t stdId, uint8_t* data)
+{
+    switch (stdId)
```

Comment:
Since the IDs are uniquely different, between motor one and motor zero. I think I'll keep this duplication because of the clarity aspect. I would need to use masks and IDs for it to work, I just think the trade off here is worth it. Especially since I really don't expect there to be more motors coming in ever, unlike the amount of MPPTs which is likely to change.